src/ngsa/app/DataAccessLayer/InMemoryDal.cs

```diff
@@ -151,120 +196,144 @@ public Task<IEnumerable<Actor>> GetActorsAsync(string q, int offset = 0, int lim
+            });
+        }
+
+        public Task<IEnumerable<string>> GetGenresAsync()
+        {
+            return Task<IEnumerable<string>>.Factory.StartNew(() => { return Genres; });
+        }
+
+        public async Task<Movie> GetMovieAsync(string movieId)
+        {
+            return await Task.Run(() =>
+            {
+                if (MoviesIndex.ContainsKey(movieId))
+                {
+                    return MoviesIndex[movieId];
+                }
+
+                throw new CosmosException("Not Found", System.Net.HttpStatusCode.NotFound, 404, string.Empty, 0);
+            }).ConfigureAwait(false);
+        }
+
+        public Task<IEnumerable<Movie>> GetMoviesAsync(MovieQueryParameters movieQueryParameters)
+        {
+            if (movieQueryParameters == null)
+            {
+                return GetMoviesAsync(string.Empty, string.Empty, offset: 0, limit: 100);
+            }
+
+            return GetMoviesAsync(movieQueryParameters.Q, movieQueryParameters.Genre, movieQueryParameters.Year, movieQueryParameters.Rating, movieQueryParameters.ActorId, movieQueryParameters.GetOffset(), movieQueryParameters.PageSize);
+        }
+
         public Task<IEnumerable<Movie>> GetMoviesAsync(string q, string genre, int year = 0, double rating = 0.0, string actorId = "", int offset = 0, int limit = 100)
         {
             List<Movie> res = new List<Movie>();
             int skip = 0;
             bool add = false;
 
-            foreach (Movie m in Movies)
+            if ((year > 0 || !string.IsNullOrEmpty(genre)) &&
+                !(year > 0 && !string.IsNullOrEmpty(genre)) &&
+                string.IsNullOrEmpty(q) &&
+                rating == 0 &&
+                string.IsNullOrEmpty(actorId) &&
+                offset == 0)
             {
-                if ((string.IsNullOrEmpty(q) || m.TextSearch.Contains(q, StringComparison.OrdinalIgnoreCase)) &&
-                    (string.IsNullOrEmpty(genre) || m.Genres.Contains(genre, StringComparer.OrdinalIgnoreCase)) &&
-                    (year < 1 || m.Year == year) &&
-                    (rating <= 0 || m.Rating >= rating))
+                if (year > 0)
                 {
-                    add = true;
-
-                    if (!string.IsNullOrEmpty(actorId))
+                    if (YearIndex.ContainsKey(year))
                     {
-                        add = false;
-
-                        actorId = actorId.Trim().ToLowerInvariant();
-
-                        foreach (Role a in m.Roles)
+                        foreach (Movie m in YearIndex[year])
                         {
-                            if (a.ActorId == actorId)
+                            res.Add(m);
+
+                            if (res.Count >= limit)
                             {
-                                add = true;
                                 break;
```

Comment:
Providing it doesn't violate a rule I'd modify that to be break; with no braces.

Reply:
it will violate fxcop rule and visual studio will change it when you run code analyze